protos/route.proto

```diff
@@ -441,6 +441,38 @@ message ResyncMirrorRequest {
 message ResyncMirrorResponse {
 }
 
+message PeerDBStateRequest {
+}
+
+enum InstanceStatus {
+  INSTANCE_STATUS_UNKNOWN = 0;
+  INSTANCE_STATUS_READY = 1;
+  INSTANCE_STATUS_MAINTENANCE = 3;
+}
+
+message InstanceInfoRequest {
+}
+
+message InstanceInfoResponse {
+  InstanceStatus status = 1;
+}
+
+enum MaintenanceStatus {
+  MAINTENANCE_STATUS_UNKOWN = 0;
+  MAINTENANCE_STATUS_START = 1;
+  MAINTENANCE_STATUS_END = 2;
+}
+
+message MaintenanceRequest {
+  MaintenanceStatus status = 1;
+  bool use_peerflow_task_queue = 2;
```

Comment:
why is this an option? can we stick to one task queue?

Reply:
a separate task queue is generally needed because we always want to run the latest version of the maintenance work flow. In some cases we may not have the maintenance task queue polling worker available, in that case flow worker + API will do it.

Reply:
In which case is that?

Reply:
Maintenance worker will only be available from the pre-upgrade (before instance gets upgraded) till end of post-upgrade (after an instance gets upgraded).